rct229/rules/section6/section6rule12.py

```diff
@@ -0,0 +1,102 @@
+                    rmrs_used=UserBaselineProposedVals(False, False, True),
+                    required_fields={
+                        "$": ["spaces", "surfaces"],
+                        "$.spaces[*]": ["interior_lighting"],
+                        "$.spaces[*].interior_lighting[*]": [
+                            "daylighting_control_type",
+                            "are_schedules_used_for_modeling_daylighting_control",
+                        ],
+                        "$.surfaces[*]": [
+                            "adjacent_to",
+                            "subsurfaces",
+                        ],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                building_p = context.proposed
+
+                daylight_flag_u = False
+                has_daylight_control_flag = False
+                for subsurface in find_all(
+                    "$..surfaces[*].subsurfaces[?classification != 'DOOR']", building_p
+                ):
+                    daylight_flag_u = True
+
+                interior_lighting_u = False
+                for lighting in find_all("$..spaces[*].interior_lighting", building_p):
+                    interior_lighting_u = lighting[0][
+                        "are_schedules_used_for_modeling_daylighting_control"
+                    ]
+                    if lighting[0]["daylighting_control_type"] != "NONE":
+                        has_daylight_control_flag = True
+
+                return {
+                    "daylight_flag_u": daylight_flag_u,
+                    "has_daylight_control_flag": has_daylight_control_flag,
+                    "interior_lighting_u": interior_lighting_u,
+                }
+
+            def manual_check_required(self, context, calc_vals=None, data=None):
+                daylight_flag_u = calc_vals["daylight_flag_u"]
+
+                return daylight_flag_u
+
+            def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+                daylight_flag_u = calc_vals["daylight_flag_u"]
+                has_daylight_control_flag = calc_vals["has_daylight_control_flag"]
+                interior_lighting_u = calc_vals["interior_lighting_u"]
+
+                if daylight_flag_u:
+                    if has_daylight_control_flag:
+                        if interior_lighting_u:
+                            manual_check_msg = MSG_WARN_DAYLIGHT
+                        else:
+                            manual_check_msg = MSG_WARN_DAYLIGHT_NO_SCHEDULE
+                    else:
+                        manual_check_msg = MSG_WARN_NO_DAYLIGHT
+
+                return manual_check_msg
+
+            def rule_check(self, context, calc_vals, data=None):
+                daylight_flag_u = calc_vals["daylight_flag_u"]
+                has_daylight_control_flag = calc_vals["has_daylight_control_flag"]
+
+                return not daylight_flag_u and not has_daylight_control_flag
```

Comment:
I think we need a `get_fail_msg()` method defined to provide the message from Case3.